Content.Server/Sericulture/SericultureSystem.cs

```diff
@@ -25,53 +18,15 @@ public override void Initialize()
     {
         base.Initialize();
 
-        SubscribeLocalEvent<SericultureComponent, ComponentInit>(OnCompInit);
-        SubscribeLocalEvent<SericultureComponent, ComponentShutdown>(OnCompRemove);
-        SubscribeLocalEvent<SericultureComponent, SericultureActionEvent>(OnSericultureStart);
         SubscribeLocalEvent<SericultureComponent, SericultureDoAfterEvent>(OnSericultureDoAfter);
     }
 
-    private void OnCompInit(EntityUid uid, SericultureComponent comp, ComponentInit args)
-    {
-        if (!_protoManager.TryIndex<InstantActionPrototype>(comp.ActionProto, out var actionProto))
-            return;
-
-        _actionsSystem.AddAction(uid, new InstantAction(actionProto), uid);
-    }
-
-    private void OnCompRemove(EntityUid uid, SericultureComponent comp, ComponentShutdown args)
-    {
-        if (!_protoManager.TryIndex<InstantActionPrototype>(comp.ActionProto, out var actionProto))
-            return;
-
-        _actionsSystem.RemoveAction(uid, new InstantAction(actionProto));
-    }
-
-    private void OnSericultureStart(EntityUid uid, SericultureComponent comp, SericultureActionEvent args)
-    {
-        if (IsHungry(uid))
-        {
-            _popupSystem.PopupEntity(Loc.GetString(comp.PopupText), uid, uid);
-            return;
-        }
-
-        var doAfter = new DoAfterArgs(uid, comp.ProductionLength, new SericultureDoAfterEvent(), uid)
-        {
-            BreakOnUserMove = true,
-            BlockDuplicate = true,
-            BreakOnDamage = true,
-            CancelDuplicate = true,
-        };
-
-        _doAfterSystem.TryStartDoAfter(doAfter);
-    }
-
     private void OnSericultureDoAfter(EntityUid uid, SericultureComponent comp, SericultureDoAfterEvent args)
     {
         if (args.Cancelled || args.Handled || comp.Deleted)
             return;
 
-        if (IsHungry(uid))
+        if (_hungerSystem.IsHungerBelowState(uid, comp.MinHungerThreshold)) // A check, just incase the doafter is somehow preformed when the entity is not in the right hunger state.
```

Comment:
This should re-validate the cost is valid or it can underflow hunger in some scenarios and should realistically be moved to shared.

Also performed not preformed.

Reply:
@metalgearsloth Sorry for asking this, but could you put this into words I can understand? Thanks.

Reply:
Nevermind I think I figured it out.

Reply:
Think I did what was asked of me.